Add note about workload identity with KPO (#2668)
astronomer/docs: astro/connect-gcp.md

To allow data pipelines running on GCP to access Google Cloud services in a secure and manageable way, Google recommends using [Workload Identity](https://cloud.google.com/kubernetes-engine/docs/concepts/workload-identity). All Astro clusters on GCP have Workload Identity enabled by default. Each Astro Deployment is associated with a Google service account that's created by Astronomer and is bound to an identity from your Google Cloud project's fixed workload identity pool.

> :::note

Reviewer: ```suggestion
:::info
```

> When connecting with Workload Identity from tasks using the Kubernetes Pod Operator (KPO), please first contact Astronomer support to enable the KPO service account binding.

Reviewer: ```suggestion
If you want to connect with Workload Identity from tasks using the KubernetesPodOperator (KPO), please first contact [Astronomer support](https://cloud.astronomer.io/support) to enable the KPO service account binding.
```
